### PR TITLE
Add messages and special actions for deprecated packages

### PR DESCRIPTION
**Pull request by benogle**
WIP

screen shot 2015-06-01 at 11 05 35 am

### CODE REVIEW

**Comment by benogle** [comment]
I'm nearly done with this. @thedaniel @kevinsawicki feel free to review. I've changed a lot.

File: lib/package-manager.coffee [hunk 1]
```diff
@@ -230,7 +231,7 @@ class PackageManager
     apmProcess = @runCommand(args, exit)
     handleProcessErrors(apmProcess, errorMessage, onError)
 
-  unload: (packageName) ->
+  unload: (name) ->
```
Comment (kevinsawicki) [inline lib/package-manager.coffee]: This is embarrassing :cry:

Comment (benogle) [inline lib/package-manager.coffee]: yeah I cant believe that was in there

Comment (benogle) [inline lib/package-manager.coffee]: I think the promise catch was eating any errors too.

File: lib/package-manager.coffee [hunk 2]
```diff
@@ -291,6 +297,26 @@ class PackageManager
...
+        if error then reject(error) else resolve()
+
+    installPromise = new Promise (resolve, reject) =>
+      @install {name: alternativePackageName}, (error) ->
+        if error then reject(error) else resolve()
+
+    Promise.all([uninstallPromise, installPromise]).then =>
+      callback(null, eventArg)
+      @emit('package-installed-alternative', eventArg)
+    .catch (error) ->
```
Comment (kevinsawicki) [inline lib/package-manager.coffee]: Does this arrow need to be `=>` since `@emit` is called?

Comment (benogle) [inline lib/package-manager.coffee]: yep

File: lib/package-card.coffee [hunk 3]
```diff
@@ -53,17 +62,20 @@ class PackageCard extends View
...
     {@name} = @pack
 
     @handlePackageEvents()
-    @handleControlsEvent(opts)
-    @updateEnablement()
+    @handleButtonEvents(options)
     @loadCachedMetadata()
 
+    @packageMessage.on 'click', 'a', (e) ->
+      href = this.getAttribute('href')
```
Comment (kevinsawicki) [inline lib/package-card.coffee]: super minor, but could be `@getAttribute` for consistency

File: lib/package-card.coffee [hunk 4]
```diff
@@ -53,17 +62,20 @@ class PackageCard extends View
...
-    @filterText = "#{@pack.name} #{owner}"
     {@name} = @pack
 
     @handlePackageEvents()
-    @handleControlsEvent(opts)
-    @updateEnablement()
+    @handleButtonEvents(options)
     @loadCachedMetadata()
 
+    @packageMessage.on 'click', 'a', (e) ->
```
Comment (kevinsawicki) [inline lib/package-card.coffee]: `e` looks unused, :fire:?

**Comment by benogle** [comment]
Ok, this is ready. Specs pass locally. This is what it looks like:

screen shot 2015-06-03 at 5 27 55 pm

**Comment by thedaniel** [comment]
LGTM, I like the addition of ListView. This package is kind of messy in general but if we refactor a bit like this with each feature addition we're in good shape.

**Pull request merged**